File: im-mobile/lib/models/read_receipt.dart
// Read Receipt Model for Flutter
class ReadReceipt {
  final int userId;
  final String conversationId;
  final String messageId;
  final int readAt;
  final List<int>? readByUsers;

  ReadReceipt({
    required this.userId,
    required this.conversationId,
    required this.messageId,
    required this.readAt,
    this.readByUsers,
  });

  factory ReadReceipt.fromJson(Map<String, dynamic> json) {
    return ReadReceipt(
      userId: json['userId'] ?? 0,
      conversationId: json['conversationId'] ?? '',
      messageId: json['messageId'] ?? '',
      readAt: json['readAt'] ?? 0,
      readByUsers: json['readByUsers'] != null
          ? List<int>.from(json['readByUsers'])
          : null,
    );
  }
}

File: im-desktop/src/types/read-receipt.ts
// Read Receipt Types
export interface ReadReceipt {
  userId: number;
  conversationId: string;
  messageId: string;
  readAt: number;
  readByUsers?: number[];
}

export interface ReadReceiptRequest {
  userId: number;
  conversationId: string;
  messageId?: string;
  messageIds?: string[];
  readAt?: number;
}
